compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -323,12 +323,18 @@ execCompile inputFile outputFile opts = withProjectRoot' (ProjectOpts Nothing (P
     withDamlIdeState opts' loggerH diagnosticsLogger $ \ide -> do
         setFilesOfInterest ide (Set.singleton inputFile)
         runAction ide $ do
-          when (optWriteInterface opts') $ do
-              mbIfaces <- writeIfacesAndHie (toNormalizedFilePath $ fromMaybe ifaceDir $ optIfaceDir opts') inputFile
-              void $ liftIO $ mbErr "ERROR: Compilation failed." mbIfaces
           mbDalf <- getDalf inputFile
           dalf <- liftIO $ mbErr "ERROR: Compilation failed." mbDalf
+          let pkgId = T.unpack $ Archive.encodePackageHash dalf
+          when (optWriteInterface opts') $ do
+              mbIfaces <-
+                  writeIfacesAndHie
+                      (Just pkgId)
+                      (toNormalizedFilePath $ fromMaybe ifaceDir $ optIfaceDir opts')
+                      inputFile
+              void $ liftIO $ mbErr "ERROR: Compilation failed." mbIfaces
           liftIO $ write dalf
+          liftIO $ putStrLn $ "Package id: " <> pkgId
```

Comment:
Why do we need this now? If we do need it, should we have it for all commands (e.g., also for `build`, …) instead of only for `compile`?